nexus-sdk/src/main/java/io/nexusrpc/handler/OperationStartResult.java

```diff
@@ -16,20 +20,26 @@ public static <R> OperationStartResult<R> sync(@Nullable R value) {
 
   /** Create a started asynchronous operation start result with the given operation ID. */
   public static <R> OperationStartResult<R> async(String operationId) {
```

Comment:
@bergundy Do you see us adding more parameters to either sync or async result? We might want to consider using a builder here instead.

cc: @cretz

Reply:
Yeah, I think we should do that here instead of adding a `links` param. What if I wanted to add "links" to my sync result? We do need to support that, e.g. for workflow signal linking which is a sync result I believe.

Reply:
Sync result will get links soon. We want to be able to link sync updates and signals.
Can't tell if / when we'll add more parameters but that could definitely happen.

Reply:
OK switched to a builder pattern here